include/picongpu/simulation/stage/ParticleBoundaries.hpp

```diff
@@ -212,26 +40,13 @@ namespace picongpu
                  *
                  * @param desc program options following boost::program_options::options_description
                  */
-                void registerHelp(po::options_description& desc)
-                {
-                    processSpecies(desc);
-                }
+                void registerHelp(boost::program_options::options_description& desc);
 
                 /** Initialize particle boundaries stage
                  *
                  * Sets boundary kind values for all affected species.
                  */
-                void init()
-                {
-                    processSpecies();
-                }
-
-            private:
-                //! Only allow customization for species with pusher
-                using SpeciesWithPusher =
-                    typename pmacc::particles::traits::FilterByFlag<VectorAllSpecies, particlePusher<>>::type;
-                //! Functor to process all affected species
-                meta::ForEach<SpeciesWithPusher, detail::ParticleBoundariesCommandLine<boost::mpl::_1>> processSpecies;
```

Comment:
Just for information, why was this moved into both the functions rather than being left here

Reply:
This avoids that this compile-time algorithm is executed even if it is not required.
The evaluation is now in the cpp because it is only there required.
This reduces the compile time.